### ISSUE TITLE
Race condition on dev branch - OSX amd64

### ISSUE THREAD

**Issue by Mzack9999** [projectdiscovery/nuclei#1309]
### Nuclei version:
2.5.4-dev

### Current Behavior:
Race condition during compilation on OSX amd64

```console
$ echo http://127.0.0.1:8000 | go run -race .

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.5.4-dev

                projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
fatal error: unexpected signal during runtime execution
[signal SIGBUS: bus error code=0x2 addr=0x7ff8309682e5 pc=0x7ff8160850ee]

runtime stack:
runtime.throw({0x53a16c7, 0x0})
        /usr/local/Cellar/go/1.17.2/libexec/src/runtime/panic.go:1198 +0x71
runtime.sigpanic()
        /usr/local/Cellar/go/1.17.2/libexec/src/runtime/signal_unix.go:719 +0x396

goroutine 50 [syscall]:
runtime.cgocall(0x500f100, 0xc0000d7ca0)
        /usr/local/Cellar/go/1.17.2/libexec/src/runtime/cgocall.go:156 +0x6e fp=0xc0000d7c78 sp=0xc0000d7c40 pc=0x400676e
net._C2func_getaddrinfo(0xc000334618, 0x0, 0xc000398630, 0xc00013a730)
        _cgo_gotypes.go:97 +0x8b fp=0xc0000d7ca0 sp=0xc0000d7c78 pc=0x439eb2b
net.cgoLookupIPCNAME.func1({0xc000334618, 0x18, 0xc00050e800}, 0x0, 0x65f4d00)
        /usr/local/Cellar/go/1.17.2/libexec/src/net/cgo_unix.go:163 +0xb2 fp=0xc0000d7d10 sp=0xc0000d7ca0 pc=0x43a17f2
net.cgoLookupIPCNAME({0x534f463, 0x3}, {0xc00050e800, 0xc0000b2600})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/cgo_unix.go:163 +0x26e fp=0xc0000d7f10 sp=0xc0000d7d10 pc=0x43a0cee
net.cgoIPLookup(0x65f4d10, {0x534f463, 0x3}, {0xc00050e800, 0x17})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/cgo_unix.go:220 +0x67 fp=0xc0000d7f80 sp=0xc0000d7f10 pc=0x43a1907
net.cgoLookupIP·dwrap·22()
        /usr/local/Cellar/go/1.17.2/libexec/src/net/cgo_unix.go:230 +0x72 fp=0xc0000d7fe0 sp=0xc0000d7f80 pc=0x43a1fb2
runtime.goexit()
        /usr/local/Cellar/go/1.17.2/libexec/src/runtime/asm_amd64.s:1581 +0x1 fp=0xc0000d7fe8 sp=0xc0000d7fe0 pc=0x4069f01
created by net.cgoLookupIP
        /usr/local/Cellar/go/1.17.2/libexec/src/net/cgo_unix.go:230 +0x1ba

goroutine 1 [select]:
net/http.(*Transport).getConn(0x656ab00, 0xc0000480c0, {{}, 0x0, {0xc00003e450, 0x5}, {0xc00050e800, 0x1b}, 0x0})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/http/transport.go:1372 +0xa45
net/http.(*Transport).roundTrip(0x656ab00, 0xc00009a200)
        /usr/local/Cellar/go/1.17.2/libexec/src/net/http/transport.go:581 +0xcdf
net/http.(*Transport).RoundTrip(0xc00009a200, 0x5ca8740)
        /usr/local/Cellar/go/1.17.2/libexec/src/net/http/roundtrip.go:18 +0x37
net/http.send(0xc00009a200, {0x5ca8740, 0x656ab00}, {0x0, 0x404fea0, 0x0})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/http/client.go:252 +0x93f
net/http.(*Client).send(0x65f5a80, 0xc00009a200, {0x43aefae, 0x14, 0x0})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/http/client.go:176 +0x158
net/http.(*Client).do(0x65f5a80, 0xc00009a200)
        /usr/local/Cellar/go/1.17.2/libexec/src/net/http/client.go:725 +0x1087
net/http.(*Client).Do(...)
        /usr/local/Cellar/go/1.17.2/libexec/src/net/http/client.go:593
github.com/projectdiscovery/nuclei-updatecheck-api/client.callRegisterServer({0x535ffab, 0x8})
        /Users/user/go/pkg/mod/github.com/projectdiscovery/nuclei-updatecheck-api@v0.0.0-20211006155443-c0a8d610a4df/client/client.go:78 +0x329
github.com/projectdiscovery/nuclei-updatecheck-api/client.GetLatestNucleiTemplatesVersion()
        /Users/user/go/pkg/mod/github.com/projectdiscovery/nuclei-updatecheck-api@v0.0.0-20211006155443-c0a8d610a4df/client/client.go:36 +0x5c
github.com/projectdiscovery/nuclei/v2/internal/runner.(*Runner).fetchLatestVersionsFromGithub(0xc0000a8090, {0xc00050e740, 0x1f})
        /Users/user/go/src/github.com/projectdiscovery/nuclei/v2/internal/runner/update.go:511 +0x50
github.com/projectdiscovery/nuclei/v2/internal/runner.(*Runner).updateTemplates(0xc0000a8090)
        /Users/user/go/src/github.com/projectdiscovery/nuclei/v2/internal/runner/update.go:80 +0x407
github.com/projectdiscovery/nuclei/v2/internal/runner.New(0x65f7b60)
        /Users/user/go/src/github.com/projectdiscovery/nuclei/v2/internal/runner/runner.go:73 +0x1ee
main.main()
        /Users/user/go/src/github.com/projectdiscovery/nuclei/v2/cmd/nuclei/main.go:26 +0x69

goroutine 5 [select]:
github.com/syndtr/goleveldb/leveldb/util.(*BufferPool).drain(0xc00016e2a0)
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/util/buffer_pool.go:206 +0x17d
created by github.com/syndtr/goleveldb/leveldb/util.NewBufferPool
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/util/buffer_pool.go:237 +0x325

goroutine 23 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc000503d40)
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db_compaction.go:90 +0x1f2
created by github.com/syndtr/goleveldb/leveldb.openDB
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db.go:142 +0x91e

goroutine 24 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain(0xc000503d40)
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db_state.go:101 +0x132
created by github.com/syndtr/goleveldb/leveldb.openDB
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db.go:143 +0x99d

goroutine 25 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc000503d40)
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db_compaction.go:825 +0x945
created by github.com/syndtr/goleveldb/leveldb.openDB
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db.go:149 +0xa54

goroutine 26 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc000503d40)
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db_compaction.go:762 +0x1dc
created by github.com/syndtr/goleveldb/leveldb.openDB
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db.go:150 +0xad8

goroutine 27 [select]:
github.com/syndtr/goleveldb/leveldb/util.(*BufferPool).drain(0xc00016e000)
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/util/buffer_pool.go:206 +0x17d
created by github.com/syndtr/goleveldb/leveldb/util.NewBufferPool
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/util/buffer_pool.go:237 +0x325

goroutine 28 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc000001ba0)
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db_compaction.go:90 +0x1f2
created by github.com/syndtr/goleveldb/leveldb.openDB
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db.go:142 +0x91e

goroutine 29 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain(0xc000001ba0)
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db_state.go:101 +0x132
created by github.com/syndtr/goleveldb/leveldb.openDB
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db.go:143 +0x99d

goroutine 30 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc000001ba0)
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db_compaction.go:825 +0x945
created by github.com/syndtr/goleveldb/leveldb.openDB
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db.go:149 +0xa54

goroutine 31 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc000001ba0)
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db_compaction.go:762 +0x1dc
created by github.com/syndtr/goleveldb/leveldb.openDB
        /Users/user/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db.go:150 +0xad8

goroutine 32 [select]:
net.(*Resolver).lookupIPAddr(0x65f4d00, {0x5cbdeb0, 0xc0000b2600}, {0x534f463, 0x0}, {0xc00050e800, 0x17})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/lookup.go:302 +0x81c
net.(*Resolver).internetAddrList(0x5cbdeb0, {0x5cbdeb0, 0xc0000b2600}, {0x534f463, 0x3}, {0xc00050e800, 0x1b})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/ipsock.go:288 +0x7c6
net.(*Resolver).resolveAddrList(0x7049178, {0x5cbdeb0, 0xc0000b2600}, {0x5350238, 0x4}, {0x534f463, 0x3}, {0xc00050e800, 0x1b}, {0x0, ...})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/dial.go:221 +0x556
net.(*Dialer).DialContext(0xc0003c5320, {0x5cbdeb0, 0xc0000b2600}, {0x534f463, 0x3}, {0xc00050e800, 0x1b})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/dial.go:406 +0x6e5
net/http.(*Transport).dial(0x656ab00, {0x5cbdeb0, 0xc0000b2600}, {0x534f463, 0x3}, {0xc00050e800, 0x1b})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/http/transport.go:1166 +0x26a
net/http.(*Transport).dialConn(0x656ab00, {0x5cbdeb0, 0xc0000b2600}, {{}, 0x0, {0xc00003e450, 0x5}, {0xc00050e800, 0x1b}, 0x0})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/http/transport.go:1604 +0xd58
net/http.(*Transport).dialConnFor(0x0, 0xc0004653f0)
        /usr/local/Cellar/go/1.17.2/libexec/src/net/http/transport.go:1446 +0x13b
created by net/http.(*Transport).queueForDial
        /usr/local/Cellar/go/1.17.2/libexec/src/net/http/transport.go:1415 +0x175

goroutine 33 [select]:
net.cgoLookupIP({0x5cbde40, 0xc000048140}, {0x534f463, 0x17}, {0xc00050e800, 0x0})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/cgo_unix.go:231 +0x289
net.(*Resolver).lookupIP(0x65f4d00, {0x5cbde40, 0xc000048140}, {0x534f463, 0x3}, {0xc00050e800, 0x17})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/lookup_unix.go:97 +0x113
net.glob..func1({0x5cbde40, 0xc000048140}, 0xc0000336a0, {0x534f463, 0x3}, {0xc00050e800, 0x17})
        /usr/local/Cellar/go/1.17.2/libexec/src/net/hook.go:23 +0x89
net.(*Resolver).lookupIPAddr.func1()
        /usr/local/Cellar/go/1.17.2/libexec/src/net/lookup.go:296 +0x133
internal/singleflight.(*Group).doCall(0x65f4d10, 0xc0003ec2d0, {0xc00050e820, 0x1b}, 0xc000048180)
        /usr/local/Cellar/go/1.17.2/libexec/src/internal/singleflight/singleflight.go:95 +0x68
created by internal/singleflight.(*Group).DoChan
        /usr/local/Cellar/go/1.17.2/libexec/src/internal/singleflight/singleflight.go:88 +0x59b
exit status 2
```

### Expected Behavior:
Race condition free

### Steps To Reproduce:
Start `simplehttpserver` then run the command:

```console
$ echo http://127.0.0.1:8000 | go run -race .
```

**Comment by Mzack9999** [comment]
Known issue of the race detector on OSX and go 1.17.x (ref. `https://github.com/golang/go/issues/41572`)

The issue is not reproducible on Linux and Windows.

**Comment by Mzack9999** [comment]
Race conditions tests implemented in #1357
